samples/system-test/detect.v2beta1.test.js

```diff
@@ -0,0 +1,148 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const test = require(`ava`);
+const {runAsync} = require('@google-cloud/nodejs-repo-tools');
+
+const cmd = 'node detect.v2beta1.js';
+//const testQuery = [`How do I sign up?`];
+const testQuery = `Where is my data stored?`;
+
+const testKnowledgeBaseName = 'TestKnowBase';
+
+let knowbaseDisplayName;
+let knowbaseFullName;
+
+let documentFullPath;
+const testDocName = `TestDoc`;
+//const testDocumentPath = `https://dialogflow.com/docs/concepts/google-projects-faq`;
+const testDocumentPath = `https://cloud.google.com/storage/docs/faq`;
+
+test.before(`It should create a knowledge base`, async t => {
+  const output = await runAsync(
+    `${cmd} createKnowledgeBase -k ${testKnowledgeBaseName}`
+  );
+  const parsedOut = output.split(`\n`);
+  const confirmName = parsedOut[1].split(`:`)[1].trim();
+  t.true(confirmName === testKnowledgeBaseName);
+  knowbaseDisplayName = confirmName;
+  knowbaseFullName = parsedOut[0].split(`:`)[1].trim();
+});
+
+test.before(`It should detect Intent with Model Selection`, async t => {
+  const output = await runAsync(`${cmd} detectIntentwithModelSelection`);
+  t.true(
+    output.includes(
+      `I can help with that. Where would you like to reserve a room?`
+    )
+  );
+});
+
+test.serial(
+  `It should register newly created knowledge base with getKnowledgeBase`,
+  async t => {
+    const output = await runAsync(
+      `${cmd} getKnowledgeBase -n "${knowbaseFullName}"`
+    );
+    //const parsedOut = output.split(`\n`);
+    t.true(
+      output.includes(knowbaseFullName) && output.includes(knowbaseDisplayName)
+    );
+  }
+);
+
+test.serial(
+  `It should locate knowledge base with listKnowledgeBases`,
+  async t => {
+    const output = await runAsync(`${cmd} listKnowledgeBases`);
+    t.true(output.includes(testKnowledgeBaseName));
+    const parsedOut = output.split(`\n`);
+    parsedOut.forEach(p => {
+      if (p.includes(testKnowledgeBaseName)) t.pass();
+    });
+  }
+);
+
+test.serial(`It should create a document in Knowledge Base`, async t => {
+  const output = await runAsync(
+    `${cmd} createDocument -n "${knowbaseFullName}" -z "${testDocumentPath}" -m "${testDocName}"`
+  );
+  t.true(output.includes(`Document created`));
+});
+
+test.serial(
+  `It should list document just created with listDocuments`,
+  async t => {
+    const output = await runAsync(
+      `${cmd} listDocuments -n "${knowbaseFullName}"`
+    );
+    const parsedOut = output.split(`\n`);
```

Comment:
❗️ **#BLOCKER**

Reply:
This is to grab document id for deletion in next test step- please advice, cannonicals use this same method, see ln 72 here: https://github.com/dialogflow/dialogflow-python-client-v2/blob/master/samples/tests/knowledge_base_test.py

Reply:
> "for deletion in next test step"

I don't understand. This test does not delete anything? What is the "next test step" that you refer to?

Reply:
`//` comments in the following snippet are simply me describing the test layout, please do not include them in your actual code :)

```js
test.serial(`It should list documents`, async t => {
  // Setup: create at least 1 knowledgebase so list shows something
  await runAsync(
    `${cmd} createKnowledgeBase -k ${testKnowledgeBaseName}`
  );

  // Now run list and verify the exact relevant output
  const output = await runAsync(
    `${cmd} listKnowledgeBases`
  );
  t.true(output.includes(`Display Name: ${knowledgeBaseDisplayName}`));
  t.true(output.includes(`Name: ${knowledgeBaseName}`));
  // The tests above are GREAT ways of verifying that output formatting is consistent across samples
  // Personally, I use tests to verify these things. I couldn't do it without tests!
  // Be sure your test asserts the CORRECT output, do not simply copy/paste the formatting from the real code
  // Instead, if this test fails because the formatting is inconsistent, keep the test, fix the code, the test should pass
  // This'll help this PR to consistently console.log(`Some Thing: ${someThing}`) instead of console.log(`Foo....${}`) etc
});
```

Note: I'm not 100% familiar with Dialogflow KnowledgeBases and what the `Name` is versus the `Display Name`. The code above simply follows common patterns for testing any `LIST` style operation. Update the actual variables and assertions as needed.

Reply:
The deleteDocument test just below...